### ISSUE TITLE
METEOR_LOCAL_DIR fails meteor test on 1.3.3+

### ISSUE THREAD

**Issue by mikeifomin** [meteor/meteor#7292]
I tried testing with METEOR_LOCAL_DIR  on a fresh meteor(1.3.3) project and it fell with an error:

```
 fs.js:439
   return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                  ^
 Error: ENOENT, no such file or directory '.meteor/local-test/shell/info.json'
     at Object.fs.openSync (fs.js:439:18)
     at Object.fs.writeFileSync (fs.js:978:15)
     at Server.<anonymous> (/Users/mike/tmp/test-app/.meteor/local-test/build/programs/server/shell-server.js:78:10)
     at Server.g (events.js:180:16)
     at Server.emit (events.js:92:17)
     at net.js:1056:10
     at process._tickCallback (node.js:458:13)
```

With Meteor 1.3.2.4 all works fine, but 1.3.3 and beyond fails

## reproduce (on OS X and Ubuntu 14.04 )

1.3.2.4 work fine
``` bash
   meteor create test-app --release=1.3.2.4 && cd test-app && meteor add practicalmeteor:mocha && METEOR_LOCAL_DIR=.meteor/local-test meteor test --driver-package practicalmeteor:mocha
```

1.3.3 **fail** 
```
   meteor create test-app --release=1.3.3 && cd test-app && meteor add practicalmeteor:mocha && METEOR_LOCAL_DIR=.meteor/local-test meteor test --driver-package practicalmeteor:mocha
```

**Comment by abernix** [comment]
`METEOR_LOCAL_DIR` was not supported in Meteor 1.3.2.4 (bfdca2aaf9df635cf7f7397719f2b7cab2596412 didn't exist) so the comment above that it worked couldn't be true.

In 1.3.3.x and 1.3.4.x, `METEOR_LOCAL_DIR` works fine, but based on the way it was written, it's necessary to specify the full path (e.g. `METEOR_LOCAL_DIR=/Users/username/your-meteor-project/.meteor/local-test`) otherwise it's probably (just guessing) trying to find it relative to `process.cwd()`.

@laosb Is this something you want to look at?

**Comment by laosb** [comment]
Yeah I'm try to look at it, but these days I'm facing my Endterm Exam. I'll do further investigation after the exam, exactly this weekend.

**Comment by mikeifomin** [comment]
@abernix thank you. With the absolute path all works as expected.

``` bash 
meteor create test-app --release=1.3.3 && cd test-app && meteor add practicalmeteor:mocha && METEOR_LOCAL_DIR=/tmp/local-test meteor test --driver-package practicalmeteor:mocha
```

**Comment by laosb** [comment]
Made a PR #7343.

**Comment by abernix** [comment]
This should have been resolved with the merging of #7343 so I will close.  Please report back if the problem still exists!
